pkg/infrastructure/azure/dns.go

```diff
@@ -0,0 +1,242 @@
+
+	useIPv6 := false
+	for _, network := range in.InstallConfig.Config.Networking.ServiceNetwork {
+		if network.IP.To4() == nil {
+			useIPv6 = true
+		}
+	}
+
+	privateRecords := []recordPrivateList{}
+	ttl := int64(300)
+	recordType := arecord
+	if useIPv6 {
+		recordType = aaaarecord
+	}
+	publicRecords := createRecordSet(fmt.Sprintf("api.%s", in.InstallConfig.Config.ObjectMeta.Name), azureTags, ttl, recordType, ipElb, "")
+	privateRecords = append(privateRecords, createPrivateRecordSet("api-int", azureTags, ttl, recordType, ipIlb, ""))
+	privateRecords = append(privateRecords, createPrivateRecordSet("api", azureTags, ttl, recordType, ipIlb, ""))
+
+	// TODO: Populate with public LB FQDN. Placeholder text as value.
+	if !private {
+		cnameRecordName := apiExternalName
+		if useIPv6 {
+			cnameRecordName = apiExternalNameV6
+		}
+		publicRecords = createRecordSet(cnameRecordName, azureTags, ttl, cname, "", in.InstallConfig.Config.ClusterDomain())
+	}
+
+	session, err := in.InstallConfig.Azure.Session()
+	if err != nil {
+		return err
+	}
+	subscriptionID := session.Credentials.SubscriptionID
+	tokenCreds, err := azidentity.NewClientSecretCredential(session.Credentials.TenantID, session.Credentials.ClientID, session.Credentials.ClientSecret, nil)
+	if err != nil {
+		return err
+	}
+	recordSetClient, err := armdns.NewRecordSetsClient(subscriptionID, tokenCreds, nil)
+	if err != nil {
+		return err
+	}
+	privateRecordSetClient, err := armprivatedns.NewRecordSetsClient(subscriptionID, tokenCreds, nil)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("zone:%s", zone)
+	logrus.Infof("private zone:%s", privatezone)
+	logrus.Infof("ip:%s", ipIlb)
+	logrus.Infof("%v", azureCluster)
+
+	// Create the records for api and api-int in the private zone and api.<clustername> for public zone.
+	// CAPI currently creates a record called "apiserver" instead of "api" so creating "api" for the installer in the private zone.
+	// TODO: check for internal publish and/or private cluster setup.
+	_, err = recordSetClient.CreateOrUpdate(ctx, baseDomainResourceGroup, zone, publicRecords.Name, publicRecords.RecordType, publicRecords.RecordSet, nil)
+	if err != nil {
+		return err
+	}
+
+	for _, record := range privateRecords {
+		_, err = privateRecordSetClient.CreateOrUpdate(ctx, resourceGroup, privatezone, record.RecordType, record.Name, record.RecordSet, nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+func createPrivateRecordSet(lbType string, azureTags map[string]*string, ttl int64, rType recordListType, ipAddress string, recordName string) (record recordPrivateList) {
+	switch rType {
+	case cname:
+		record = recordPrivateList{
+			Name:       lbType,
+			RecordType: armprivatedns.RecordTypeCNAME,
+			RecordSet: armprivatedns.RecordSet{
+				Properties: &armprivatedns.RecordSetProperties{
+					CnameRecord: &armprivatedns.CnameRecord{
+						Cname: &recordName,
+					},
+					TTL:      &ttl,
+					Metadata: azureTags,
+				},
+			},
+		}
```

Comment:
The only thing that varies in these switch statements are `RecordType` and `Properties`, so those should be the values set in cases:

```suggestion
        var recordType armprivatedns.RecordType
        properties := &armprivatedns.RecordSetProperties{
					TTL:      &ttl,
					Metadata: azureTags,
				},
	switch rType {
	case cname:
		recordType = armprivatedns.RecordTypeCNAME
		properties.CnameRecord = &armprivatedns.CnameRecord{
						Cname: &recordName,
					},
```
Then return the `recordPrivateList` at the end of the function with those values plugged in